### PR TITLE
lua: Update to 5.4.2 and remove winpty stuff fixing as per ag/the_sil…

### PR DESCRIPTION
**Pull request by mingwandroid**
…ver_surfer

With this fix, luac can compile source code from stdin.

### CODE REVIEW

**Comment by lazka** [comment]
Since this is an ABI break I guess we need to rebuild some things from https://packages.msys2.org/package/mingw-w64-x86_64-lua?repo=mingw64 (Required B)?

**Comment by mingwandroid** [comment]
Yeah, I think so @lazka. I only tested with VLC plugins so far.

**Comment by mingwandroid** [comment]
.. and I did have to rebuild VLC for that since our current builds do not use lua.

**Comment by lazka** [comment]
Ideally you would bump the pkgrel of all affected packages in this PR. ABI breaks aren't fun...

**Comment by mingwandroid** [comment]
Sure, I can go ahead and do that.

File: mingw-w64-lua/0001-When-running-Win32-version-of-luac-on-mintty-or-othe.patch [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+ 
+ #if !defined(LUA_PROGNAME)
+ #define LUA_PROGNAME		"lua"
+@@ -365,7 +366,11 @@ static int handle_luainit (lua_State *L) {
+ #if defined(LUA_USE_POSIX)	/* { */
+ 
+ #include <unistd.h>
++#ifdef _WIN32
++#define lua_stdin_is_tty()	iscygpty(0)
++#define
```
Comment (jeremyd2019) [inline mingw-w64-lua/0001-When-running-Win32-version-of-luac-on-mintty-or-othe.patch:295]:
```suggestion
+#else
```

**Review by jeremyd2019 — CHANGES_REQUESTED** [review]

**Comment by mingwandroid** [comment]
Thanks @jeremyd2019

**Comment by jeremyd2019** [comment]
sha256sums?

**Comment by mingwandroid** [comment]
Gah!

**Comment by mingwandroid** [comment]
I'm still rusty!

**Comment by jeremyd2019** [comment]
looks like luarocks didn't like lua 5.4

```
2021-02-18T17:16:17.7494964Z ==> Starting build()...
2021-02-18T17:16:18.0638628Z 
2021-02-18T17:16:18.0641778Z Configuring LuaRocks version 3.4.0...
2021-02-18T17:16:18.0643291Z 
2021-02-18T17:16:18.1350753Z Lua 5.3 interpreter not found in $PATH
2021-02-18T17:16:18.1610615Z You may want to use the flags --with-lua, --with-lua-bin and/or --lua-version
2021-02-18T17:16:18.1698936Z Run ./configure --help for details.
2021-02-18T17:16:18.1706282Z 
2021-02-18T17:16:18.1714082Z configure failed
2021-02-18T17:16:18.1723976Z 
2021-02-18T17:16:18.2279977Z ==> ERROR: A failure occurred in build().
2021-02-18T17:16:18.2552793Z     Aborting...
```

**Comment by lazka** [comment]
Seems we are hitting https://github.com/mltframework/mlt/issues/636

**Comment by mingwandroid** [comment]
Yeah, I looked at what linux distros are doing about this and they either drop the opencv dependency or make it link  to an older opencv. From first glance, this shouldn't be too bad to patch though. Thoughts?

**Comment by mingwandroid** [comment]
Merged? Cool, what about lua-rocks? On the back-burner? Thanks @lazka!

**Comment by lazka** [comment]
There is no movement upstream, so I disabled opencv for now.

**Pull request merged**